test/integration_test.py

```diff
@@ -386,13 +386,21 @@ def test_catalog_listing(self):
 
     def test_snapshot_listing(self):
         """
-        Test with a small page size to be sure paging works
+        Test with minimal page sizes (1, 2) and a subset of deployment-specific
+        snapshots (obtain via a filtered request). To make sure paging works.
```

Comment:
Too much detail IMO. I moved the 2nd half to an inline comment.

```suggestion
        Test with small page sizes to make sure paging works.
```